clients.go

```diff
@@ -18,10 +18,10 @@ import (
 )
 
 type ClientsConfig struct {
-	// AddToScheme is an optional way to extend the known types to the gobal
+	Logger micrologger.Logger
+	// SchemeBuilder is an optional way to extend the known types to the global
 	// client-go scheme. Make use of it for custom CRs.
-	AddToScheme func(*runtime.Scheme) error
-	Logger      micrologger.Logger
+	SchemeBuilder runtime.SchemeBuilder
```

Comment:
That doesn't prevent from importing `k8s.io/apimachinery/pkg/runtime` in the using code.

The original type is defined like (https://godoc.org/k8s.io/apimachinery/pkg/runtime#SchemeBuilder):

```go
type SchemeBuilder []func(*Scheme) error
```

I'd suggest defining identical type in this package and then replacing line 100 with:

```diff
- err = config.SchemeBuilder.AddToScheme(scheme.Scheme)
+ err = runtime.SchemeBuilder(config.SchemeBuilder).AddToScheme(scheme.Scheme)
```

Then we can use `k8sclient.SchemeBuilder` and no additional import is required.

Reply:
I don't have any objections for this, but I also don't fully understand why importing `runtime` is a problem?

Reply:
IMO it's easier to deal with backward compatibility if things change again when you don't expose too much. For me personally is less thinking. Also it feels more complete.

Reply:
Yes it's a close one but keeping the imports slimmer makes sense to me. I'll update to go with this.